### ISSUE TITLE
事前学習言語モデルの動向 / Survey of Pretrained Language Models

### ISSUE THREAD

**Issue by AkihikoWatanabe** [AkihikoWatanabe/paper_notes#329]
https://speakerdeck.com/kyoun/survey-of-pretrained-language-models

**Comment by AkihikoWatanabe** [comment]
[2019/06まで]
・ELMo（双方向2層LSTM言語モデル）
・GPT（left-to-rightの12層Transformer自己回帰言語モデル）
・BERT（24層のTransformer双方向言語モデル）
・MT-DNN（BERTの上にマルチタスク層を追加した研究）
・XLM（パラレル翻訳コーパスを用いてクロスリンガルに穴埋めを学習）
・TransformerXL（系列長いに制限のあった既存モデルにセグメントレベルの再帰を導入し長い系列を扱えるように）
・GPT-2（48層Transformerの自己回帰言語モデル）
・ERNIE 1.0（Baidu, エンティティとフレーズの外部知識を使ってマスクに利用）
・ERNIE（Tsinghua, 知識グラフの情報をfusionしたLM）
・Glover（ドメイン、日付、著者などを条件とした生成を可能としたGPT）
・MASS（Encoder-Decoder型の生成モデルのための事前学習）
・UniLM（Sequence-to-Sequenceを可能にした言語モデル）
・XLNet（自己回帰（単方向）モデルと双方向モデルの両方の利点を得ることを目指す）

[2019/07~]
・SpanBERT（i.i.dではなく範囲でマスクし、同時に範囲の境界も予測する）
・ERNIE 2.0（Baidu, マルチタスク事前学習; 単語レベル・構造レベル・意味レベル）
・RoBERTa（BERTと同じ構造で工夫を加えることで性能向上）
　- より大きなバッチサイズを使う（256から8192）
　- より多くのデータを使う（16GBから160GB）
　- より長いステップ数の学習をする（BERT換算で16倍）
　- 次文予測（NSP）は不要
　→ GLUEでBERT, XLNetをoutperform
・StructBERT (ALICE, NSPに代わる学習の目的関数を工夫)
　- マスクした上で単語の順番をシャッフルし元に戻す
　- ランダム・正順・逆順の3種類を分類
　→ BERTと同サイズ、同データでBERT, RoBERTa超え
・DistilBERT（蒸留により、12層BERTを6層に小型化（40%減））
　- BERTの出力を教師として、生徒が同じ出力を出すように学習
　- 幅（隠れ層）サイズを減らすと、層数を経あｒスよりも悪化
　→ 推論は60%高速化、精度は95%程度を保持
・Q8BERT（精度を落とさずにfine-tuning時にBERTを8bit整数に量子化）
　- Embedding, FCは8bit化、softmax, LNorm, GELUは32bitをキープ
　→ モデルサイズ1/4, 速度3.7倍
・CTRL（条件付き言語モデル）
　- 条件となる制御テキストを本文の前に与えて学習
　- 48層/1280次元Transformer（パラメータ数1.6B）
・MegatronLM（72層、隠れ状態サイズ3072、長さ1024; BERTの24倍サイズ）
・ALBERT（BERTの層のパラメータをすべて共有することで学習を高速化; 2020年あたりのデファクト）
　- Largeを超えたモデルは学習が難しいため、表現は落ちるが学習しやすくした
　- 単語埋め込みを低次元にすることでパラメータ数削減
　- 次文予測を、文の順序入れ替え判定に変更
　→ GLUE, RACE, SQuADでSoTAを更新
・T5（NLPタスクをすべてtext-to-textとして扱い、Enc-Dec Transformerを745GBコーパスで事前学習して転移する）
　- モデルはEncoder-DecoderのTransformer
　- 学習タスクをエンコーダ・デコーダに合わせて変更
　- エンコーダ側で範囲を欠落させて、デコーダ側で予測
　→ GLUE, SuperGLUE, SQuAD1.1, CNN/DMでSoTA更新
・BART（Seq2Seqの事前学習として、トークンマスク・削除、範囲マスク、文の入れ替え、文書の回転の複数タスクで学習）
　→ CNN/DMでT5超え、WMT'16 RO-ENで逆翻訳を超えてSoTA

**Comment by AkihikoWatanabe** [comment]
ELMo, GPT, BERT, GPT-2, XLNet, RoBERTa, DistilBERT, ALBERT, T5あたりは良く見るような感

**Comment by AkihikoWatanabe** [comment]
各データセットでの各モデルの性能も後半に記載されており興味深い。

ちなみに、CNN/DailyMail Datasetでは、T5, BARTあたりがSoTA。
R2で比較すると
　- Pointer-Generator + Coverage Vectorが17,28
　- LEAD-3が17.62
　- BARTが21.28
　- T5が21.55
となっている
